src/gluonts/torch/modules/piecewise_linear.py

```diff
@@ -0,0 +1,264 @@
+        knots: torch.Tensor,
+        validate_args=False,
+    ) -> None:
+        self.gamma, self.slopes, self.knots = gamma, slopes, knots
+        self.m, self.knots_pos = PiecewiseLinear._to_orig_params(slopes, knots)
+
+        batch_shape = self.gamma.shape
+        super(PiecewiseLinear, self).__init__(
+            batch_shape=batch_shape, validate_args=validate_args
+        )
+
+    @staticmethod
+    def _to_orig_params(
+        slopes: torch.Tensor, knots: torch.Tensor
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        m = PiecewiseLinear.parametrize_slopes(slopes)
+        knots_pos = PiecewiseLinear.parametrize_knots(knots)
+        return m, knots_pos
+
+    @staticmethod
+    def parametrize_slopes(slopes: torch.Tensor) -> torch.Tensor:
+        slopes_parametrized = slopes
+        slopes_parametrized[..., 1:] = torch.diff(slopes, dim=-1)
+        return slopes_parametrized
+
+    @staticmethod
+    def parametrize_knots(knots: torch.Tensor) -> torch.Tensor:
+        knots_pos = torch.cumsum(knots, dim=-1)[
+            ..., :-1
+        ]  # the last entry is 1 and can be omitted
+        return knots_pos
+
+    def quantile_internal(
+        self, u: torch.Tensor, dim: Optional[int] = None
+    ) -> torch.Tensor:
+        # output shape = u.shape
+        if dim is not None:
+            # when num_samples!=None
+            #
+            # In testing
+            # dim = 0
+            # u.shape          = (num_samples, *batch_shape)
+            # gamma.shape      = (1, *batch_shape)
+            # other_para.shape = (1, *batch_shape, num_knots)
+            #
+            # In training, u_tilde is needed to compute CRPS
+            # dim = -2
+            # u.shape          = (*batch_shape, num_knots)
+            # gamma.shape      = (*batch_shape, 1)
+            # other_para.shape = (*batch_shape, 1, num_knots)
+
+            gamma = self.gamma.unsqueeze(dim=0 if dim == 0 else -1)
+            knots_pos, m = self.knots_pos.unsqueeze(dim=dim), self.m.unsqueeze(
+                dim=dim
+            )
+        else:
+            # when num_sample is None
+            #
+            # In testing
+            # dim = None
+            # u.shape          = (*batch_shape)
+            # gamma.shape      = (*batch_shape)
+            # other_para.shape = (*batch_shape, num_knots)
+
+            gamma = self.gamma
+            knots_pos, m = self.knots_pos, self.m
+
+        u = u.unsqueeze(-1)
+        u_spline = F.relu(u - knots_pos)
+        quantile = gamma + torch.sum(m * u_spline, dim=-1)
+
+        return quantile
+
+    def quantile(self, u: torch.Tensor) -> torch.Tensor:
+        return self.quantile_internal(u, dim=0)
+
+    def get_u_tilde(self, z: torch.Tensor) -> torch.Tensor:
+        """
+        compute the quantile levels u_tilde s.t. quantile(u_tilde)=z
+
+        Input
+        z: observations, shape = gamma.shape = (*batch_size)
+
+        Output
+        u_tilde: of type torch.Tensor
+        """
```

Comment:
The docs build logs say
```
Warning, treated as error:
/home/runner/work/gluon-ts/gluon-ts/src/gluonts/torch/modules/piecewise_linear.py:docstring of gluonts.torch.modules.piecewise_linear.PiecewiseLinear.get_u_tilde:3:Inline emphasis start-string without end-string.
```

So it looks like they don't like this docstring. You can try to use the same formatting as other docstrings in the package:

```
"""
<Short description>

<Optional long description>

Parameters
----------
<arg-name-1>
    <description>
<arg-name-2>
    <description>
...
Returns
-------
<return type>
    <description>
```

and wrapping any python code in double-backticks (`)